internal/dependency/manager.go

```diff
@@ -243,3 +242,14 @@ func (dm *DependendcyManager) getVersions(name string) ([]*semver.Version, error
 		return parsedVersions, nil
 	}
 }
+
+func (dm *DependendcyManager) getManifest(ctx context.Context, pkg ctrlpkg.Package) (*v1alpha1.PackageManifest, error) {
+	if pi, err :=
+		dm.pkgClient.GetPackageInfo(ctx, names.PackageInfoName(pkg)); err != nil && !apierrors.IsNotFound(err) {
+		return nil, err
+	} else if (err != nil && apierrors.IsNotFound(err)) || (err == nil && pi.Status.Manifest == nil) {
```

Comment:
`IsNotFound(err)` implies `err != nil`

```suggestion
	} else if apierrors.IsNotFound(err) || (err == nil && pi.Status.Manifest == nil) {
```